pkg/apis/nfd/nodefeaturerule/expression.go

```diff
@@ -122,40 +122,94 @@ func evaluateMatchExpression(m *nfdv1alpha1.MatchExpression, valid bool, value i
 			if len(m.Value) != 1 {
 				return false, fmt.Errorf("invalid expression, 'value' field must contain exactly one element for Op %q (have %v)", m.Op, m.Value)
 			}
-
-			l, err := strconv.Atoi(value)
-			if err != nil {
-				return false, fmt.Errorf("not a number %q", value)
-			}
-			r, err := strconv.Atoi(m.Value[0])
-			if err != nil {
-				return false, fmt.Errorf("not a number %q in %v", m.Value[0], m)
+			if m.Type != "" && m.Type != nfdv1alpha1.TypeVersion {
+				return false, fmt.Errorf("invalid expression, 'type' field only accepts %q value for Op %q (have %q)", nfdv1alpha1.TypeVersion, m.Op, m.Type)
```

Comment:
I'd suggest to add this one as the default: case below

Reply:
Fixed.